skglm/solvers/cd_solver.py

```diff
@@ -144,10 +152,27 @@ def cd_solver_path(X, y, datafit, penalty, alphas=None,
                 w = np.zeros(n_features, dtype=X.dtype)
                 Xw = np.zeros(X.shape[0], dtype=X.dtype)
 
-        sol = cd_solver(
-            X, y, datafit, penalty, w, Xw,
-            max_iter=max_iter, max_epochs=max_epochs, p0=p0, tol=tol,
-            use_acc=use_acc, verbose=verbose, ws_strategy=ws_strategy)
+        if (isinstance(datafit, (Quadratic, Quadratic_32)) and n_samples > n_features
+                and n_features < 10_000) or solver in ("cd_gram", "fista"):
+            # Gram matrix must fit in memory hence the restriction n_features < 1e5
+            if not isinstance(datafit, (Quadratic, Quadratic_32)):
```

Comment:
I think this bit is unreachable because the check is already performed L155

Reply:
I've placed it because the first condition is "`isinstance....` OR solver in ...". If the user manually inputs "cd_gram", I want to catch a wrong datafit, hence L158. Overkill maybe? Should we even expose `solver`? I think it is convenient for benchmarks. 
WDYT?

Reply:
Ok I understood, thanks. 
Maybe we can indent the first if, breaking line before  `or solver` to make it more visible ?

Reply:
I tried to make it more obvious. WDYT?